alpha/lib/model/DeliveryProfileLiveAppleHttp.php

```diff
@@ -120,13 +120,61 @@ protected function isDvrContent($content) {
 	}
 	
 	/**
-	 * Build all streaming flavors array
+	 * Fetch the manifest from the remote Wwoza server and build all flavors array
 	 * @param string $url
 	 */
-	private function buildM3u8Flavors($url, array &$flavors, array $kLiveStreamParamsArray, $flavorBitrateInfo = array())
+	private function buildProxiedFlavorsArray($url, array &$flavors, $doaminPrefix)
 	{
-		$domainPrefix = $this->getDeliveryServerNodeUrl(true);
+		if($this->getDisableExtraAttributes())
+			$url = kDeliveryUtils::addQueryParameter($url, "attributes=off");
+		
+		$manifest = KCurlWrapper::getContent($url);
+		if(!$manifest)
+			return;
 		
+		$manifestLines = explode("\n", $manifest);
+		$manifestLine = reset($manifestLines);
+		while($manifestLine)
+		{
+			$lineParts = explode(':', $manifestLine, 2);
+			if($lineParts[0] === '#EXT-X-STREAM-INF')
+			{
+				// passing the url as urlPrefix so that only the path will be tokenized
+				$flavor = array(
+						'url' => '',
+						'urlPrefix' => requestUtils::resolve(next($manifestLines), $url),
+						'domainPrefix' => $domainPrefix,
+						'ext' => 'm3u8',
+				);
+		
+				$attributes = explode(',', $lineParts[1]);
+				foreach($attributes as $attribute)
+				{
+					$attributeParts = explode('=', $attribute, 2);
+					switch($attributeParts[0])
+					{
+						case 'BANDWIDTH':
+							$flavor['bitrate'] = $attributeParts[1] / 1024;
+							break;
+								
+						case 'RESOLUTION':
+							list($flavor['width'], $flavor['height']) = explode('x', $attributeParts[1], 2);
+							break;
+					}
+				}
+				$flavors[] = $flavor;
+			}
+				
+			$manifestLine = next($manifestLines);
```

Comment:
The parsing is according to HLS and Wowoza standards and it worked for us on production from the beginning so I don't think changing it would be the right thing to do.